docs/dev/framework/caching.md

```diff
@@ -217,24 +217,104 @@ Moreover, you don't have to register to all the different callbacks such as `ons
-_fragment controllers_. These fragments are then rendered with their defined renderer 
-and merged into the main content. Each fragment can also provide its own response
-and thus define whether it can be cached or not. If a fragment returns a response
-with a `Cache-Control: private` header for example, then the page on which the fragment
-is visible cannot be cached. On the other hand, if the fragment can be cached, it
-can provide its own cache tags, as mentioned previously.
-
-Contao brings its own `forward` fragment renderer, which provides the fragment with 
-a full clone of the request. This provides the fragment with the full `POST` data
-for example, contrary to Symfony's default `inline` renderer.
-
-The renderer can also be set to `esi`. In that case, if Symfony detects that it 
-is talking to  a gateway cache that supports ESI (like Symfony's built in reverse 
-proxy, that Contao uses), it generates an ESI include tag. See also [Symfony's documentation][esi] 
-on _Edge Side Includes_.
+_fragment controllers_. These fragments are then rendered with their defined renderer
+and merged into the main content. Each fragment returns a response and thus can
+tell whether it can be cached or not.
+
+
+There are two fundamentally different ways to render fragments:
+ 1. **Inline:** Inside the main page request and merged with the content before caching. In Symfony, this is handled
+    by the `inline` renderer, but Contao provides its own default `forward` renderer, which copies the original
+    request (whereas Symfony does not) to e.g. pass POST data to the subrequest.
+ 2. **Subrequest:** By generating an (internal) URL for the fragment, the fragment content is merged **after** caching
+    the page, which allows a fragment to have a different cache time than the main page. The `esi` renderer tells
+    the reverse proxy (cache) to fetch multiple URLs on a request and merge fragments before delivering to the client.
+    There's also an `hinclude` renderer which allows browser to merge fragments using JavaScript, but the client-side
+    part is not provided out-of-the-box.
+
+
+### Inline Fragments
+
+{{% notice note %}}
+Before **Contao 4.9**, inline fragments cannot affect the cache time of the page response.
+{{% /notice %}}
+
+By default, fragments like frontend modules and content elements are rendered directly inside the main request.
+This means the content of an inline fragment will be cached for as long as the page is cached, as configured
+in the page setting. If a fragment returns caching information in its response, the page's cache time and the fragment
+will be merged to the lowest common denominator. For example, if the page is cacheable for a day, but the fragment
+only for one hour, the whole page will only be cached for one hour.
+
+A more common use case would be something like a `{{date::Y}}` insert tag, which renders the current year. This
```

Comment:
I don't think it's a good idea to take something as an example which does the exact opposite. `{{date::Y}}` is rendered as ESI at the moment and if I were to read this article I would immediately go and see how this example is technically implemented only to see that it's not the case. That's very confusing.

Reply:
I kinda agree, but that would mean we can't provide an example, because the feature is not used by the core yet. I thought about start/stop times of content elements, but that's not implemented either.

Reply:
Then maybe add a hint that this is an example and is still a TODO in Contao core? wdyt @fritzmg

Reply:
Hmm.. may be create a fictional example with an hypothetical `{{my_date::Y}}` Insert Tag, to not allure to Contao's own `{{date}}` Insert Tag (wich is simply a special case)? I think that would be good, because currently I wouldn't even know how to create an Insert Tag that would be rendered as ESI ;)

Reply:
You can't. It's hardcoded which is why `my_date` wouldn't work :) That's why I'm saying. Andy is taking an example of something we want to do in future but doesn't work at all today.

Reply:
I see :). Well then may be simply use `MyDateElementController` or something as an example, which simply returns `date('Y')` as the response's content?